src/js/components/FrameworkConfiguration.js

```diff
@@ -0,0 +1,381 @@
+import React, { PropTypes, Component } from "react";
+import { Confirm } from "reactjs-components";
+import deepEqual from "deep-equal";
+
+import FullScreenModal from "#SRC/js/components/modals/FullScreenModal";
+import FullScreenModalHeader
+  from "#SRC/js/components/modals/FullScreenModalHeader";
+import FullScreenModalHeaderActions
+  from "#SRC/js/components/modals/FullScreenModalHeaderActions";
+import FullScreenModalHeaderTitle
+  from "#SRC/js/components/modals/FullScreenModalHeaderTitle";
+import FullScreenModalHeaderSubTitle
+  from "#SRC/js/components/modals/FullScreenModalHeaderSubTitle";
+import ToggleButton from "#SRC/js/components/ToggleButton";
+import HashMapDisplay from "#SRC/js/components/HashMapDisplay";
+import ModalHeading from "#SRC/js/components/modals/ModalHeading";
+import UniversePackage from "#SRC/js/structs/UniversePackage";
+import Icon from "#SRC/js/components/Icon";
+import Util from "#SRC/js/utils/Util";
+import StringUtil from "#SRC/js/utils/StringUtil";
+import CosmosErrorMessage from "#SRC/js/components/CosmosErrorMessage";
+import FrameworkConfigurationForm
+  from "#SRC/js/components/FrameworkConfigurationForm";
+
+export default class FrameworkConfiguration extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      reviewActive: props.isInitialDeploy,
+      focusFieldPath: this.getFirstFieldFocusPath(),
+      jsonEditorActive: false,
+      hasChangesApplied: false,
+      isConfirmOpen: false,
+      isOpen: true
+    };
+  }
+
+  onFormDataChange(formData) {
+    this.props.onFormDataChange(formData);
+
+    this.setState({ hasChangesApplied: true });
+  }
+
+  onFormErrorChange(formErrors) {
+    this.props.onFormErrorChange(formErrors);
+  }
+
+  onFocusFieldPathChange(focusFieldPath) {
+    if (deepEqual(focusFieldPath, this.state.focusFieldPath)) {
+      return;
+    }
+
+    this.setState({ focusFieldPath });
+  }
+
+  onActiveTabChange(activeTab) {
+    const { focusFieldPath } = this.state;
+    const { packageDetails } = this.props;
+    const currentActiveTab = focusFieldPath[0];
+
+    if (deepEqual(activeTab, currentActiveTab)) {
+      return;
```

Comment:
Explicitly `return false` instead relying on engine coercion which will be undefined in this case.